### ISSUE TITLE
Open source contributing: 28 July

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#513]
## Prepare gocritic

```bash
go get -u github.com/go-critic/go-critic/...
```

For this event, it's convenient to have these:
```bash
export STABLE_CHECKERS='appendCombine,builtinShadow,captLocal,flagDeref,ifElseChain,paramTypeCombine,rangeExprCopy,rangeValCopy,singleCaseSwitch,switchTrue,typeSwitchVar,typeUnparen,underef,unslice'
export EXPERIMENTAL_CHECKERS='appendAssign,boolExprSimplify,caseOrder,defaultCaseOrder,dupBranchBody,dupSubExpr,namedConst,regexpMust,sloppyLen'
export ALL_CHECKERS="${STABLE_CHECKERS},${EXPERIMENTAL_CHECKERS}"
```

With `ALL_CHECKERS`, it's possible to run `check-project` with all (non-opinionated) checkers:
```bash
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/project/import/path
```

### [prometheus](https://github.com/prometheus/prometheus)

```bash
go get -v -u github.com/prometheus/prometheus/...
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/github.com/prometheus/prometheus/
# should emit warnings
```

Test before pushing:
```bash
cd $GOPATH/src/github.com/prometheus/prometheus
make test
```

### [go-reform](https://github.com/go-reform/reform)

Requires `sqlite3`. (`sudo apt-get install sqlite3`, etc.)

```bash
go get -u gopkg.in/reform.v1/...
go get -u syreclabs.com/go/faker
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/gopkg.in/reform.v1
# should emit warnings
```

Test before pushing:
```bash
cd $GOPATH/src/gopkg.in/reform.v1
make deps
make test
make test-dc
```

### [golangci-lint](https://github.com/golangci/golangci-lint)

```go
# ensure that $GOPATH/bin is under $PATH
go get github.com/golangci/golangci-lint/...
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/github.com/golangci/golangci-lint
# should emit warnings
```

Test before pushing:
```bash
cd $GOPATH/src/github.com/golangci/golangci-lint/
make check_generated test
```

### [go-graphql](https://github.com/graphql-go/graphql)

```bash
go get github.com/graphql-go/graphql/...
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/github.com/graphql-go/graphql
# should emit warnings
```

Test before pushing:
```bash
go test -v github.com/graphql-go/graphql
```

### [sqlx](https://github.com/jmoiron/sqlx)

Notice: PR review/merge speed is low.

```bash
go get github.com/jmoiron/sqlx
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/github.com/jmoiron/sqlx
# should emit warnings
```

Test before pushing:
```bash
go test -v github.com/jmoiron/sqlx
```

### [testify](https://github.com/stretchr/testify)

```bash
go get -u github.com/stretchr/testify/...
gocritic check-project -enable=$ALL_CHECKERS $GOPATH/src/github.com/stretchr/testify
# should emit warnings
```

Test before pushing:
```bash
cd $GOPATH/src/github.com/stretchr/testify/
go test -v -race $(go list ./... | grep -v vendor)
```

**Comment by Quasilyte** [comment]
@cristaloleg, do you want to join us remotely? :)

**Comment by AlekSi** [comment]
For reform it's recommended to use `make deps` to install dependencies, and `make test-dc` to run all tests (both unit and integrational with Docker Compose).

**Comment by Quasilyte** [comment]
https://github.com/jmoiron/sqlx/pull/434

**Comment by Quasilyte** [comment]
Mission accomplished.
Thanks to everyone for cooperation.

![image](https://user-images.githubusercontent.com/6286655/43360739-25160b68-92c5-11e8-9b97-c27b573cea4f.png)
